source/LoadPanel.cpp

```diff
@@ -51,35 +51,34 @@ this program. If not, see <https://www.gnu.org/licenses/>.
 using namespace std;
 
 namespace {
-	// Return a pair containing settings to use for time formatting.
-	pair<const char*, const char*> TimestampFormatString(Preferences::DateFormat format)
+	// Return a format string containing settings to use for time formatting.
+	const char * TimestampFormatString(Preferences::DateFormat format)
 	{
-		// pair<string, string>: Linux (1st) and Windows (2nd) format strings.
 		switch(format)
 		{
 			case Preferences::DateFormat::YMD:
-				return make_pair("%F %T", "%F %T");
+				return "%F %T";
 			case Preferences::DateFormat::MDY:
-				return make_pair("%-I:%M %p on %b %-d, %Y", "%#I:%M %p on %b %#d, %Y");
+				return "%I:%M %p on %b %d, %Y";
 			case Preferences::DateFormat::DMY:
 			default:
-				return make_pair("%-I:%M %p on %-d %b %Y", "%#I:%M %p on %#d %b %Y");
+				return "%I:%M %p on %d %b %Y";
```

Comment:
According to this page: https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/strftime-wcsftime-strftime-l-wcsftime-l?view=msvc-170
The `#` is used to indicate that leading zeroes should be removed.
I don't think we should be changing that.
I'm guessing the `-` indicates to remove these leading zeroes in gcc.

Given that the reported problem seems to be limited to Linux users and Windows binaries compiled with gcc, I would guess the problem is in the first string in these pairs, not the second.

Building with gcc and launching in a debugger on Windows, when I step through this method, it still runs the "_WIN32" code, so I'm going to guess the issue is that the format string that Microsoft wants is being passed into the gcc implementation, so while the Microsoft runtime takes `%#I` and `%#d` to mean "hour without leading zero" and "day of the month without leading zero," the gcc implementation discards the "%#" and then just copies the plaintext "I" and "d".

I note that these timestamp strings used to be generated by strftime. Is it possible that the `-` specifier is supported there but not in put_time?

This page: https://man7.org/linux/man-pages/man3/strftime.3.html
Suggests that, with some extensions, it is supported. Is that what has changed? Was the gcc version previously compiled with an appropriate extension that provides this functionality but no longer is?

Reply:
Both links you gave are for strftime, which did not have problems with # or -.The program was changed to std::put_time, which is what caused the issue.

As I see it, the options are either:
a) Use strftime
b) Have a 0 in front of the time/day

Reply:
I prefer a.